python/pyspark/sql/connect/session.py

```diff
@@ -183,11 +179,17 @@ def getOrCreate(self) -> "SparkSession":
 
             if has_channel_builder:
                 assert self._channel_builder is not None
-                _active_spark_session = SparkSession(connection=self._channel_builder)
+                return SparkSession(connection=self._channel_builder)
             else:
                 spark_remote = to_str(self._options.get("spark.remote"))
                 assert spark_remote is not None
-                _active_spark_session = SparkSession(connection=spark_remote)
+                return SparkSession(connection=spark_remote)
+
+        def getOrCreate(self) -> "SparkSession":
+            global _active_spark_session
+            if _active_spark_session is not None:
+                return _active_spark_session
+            _active_spark_session = self.create()
             return _active_spark_session
```

Comment:
Wouldn't it be better if `getOrCreate()` operated as would be expected from the API with such a name?

That is, create a new session when a new URL is provided and return an existing session when a previous URL is provided.
(Either cache with the URL as the key or spark_opts as the key.)

`_active_spark_session` can either be first in or last in.

Reply:
While I don't disagree with the observation, this would be a change to the current implementation and should not be part of this PR.

Reply:
That's fair.